cmd/kubectl-gadget/collector.go

```diff
@@ -0,0 +1,389 @@
+			"threads",
+			"t",
+			false,
+			fmt.Sprintf("Show all threads"),
+		)
+	}
+}
+
+func init() {
+	gadgetv1alpha1.AddToScheme(scheme.Scheme)
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func collectorCmdRun(subCommand string) func(*cobra.Command, []string) {
+	return func(cmd *cobra.Command, args []string) {
+		contextLogger := log.WithFields(log.Fields{
+			"command": fmt.Sprintf("kubectl-gadget %s", subCommand),
+			"args":    args,
+		})
+
+		traceID, err := randomTraceID()
+		if err != nil {
+			contextLogger.Fatalf("Error while generating random trace id: %s", err)
+		}
+
+		client, err := k8sutil.NewClientsetFromConfigFlags(KubernetesConfigFlags)
+		if err != nil {
+			contextLogger.Fatalf("Error in creating setting up Kubernetes client: %q", err)
+		}
+
+		labelsSelector := map[string]string{}
+		if collectorParamLabel != "" {
+			pairs := strings.Split(collectorParamLabel, ",")
+			for _, pair := range pairs {
+				kv := strings.Split(pair, "=")
+				if len(kv) != 2 {
+					contextLogger.Fatalf("labels should be a comma-separated list of key-value pairs (key=value[,key=value,...])\n")
+				}
+				labelsSelector[kv[0]] = kv[1]
+			}
+		}
+
+		namespace := ""
+		if !collectorParamAllNamespaces {
+			namespace, _, _ = KubernetesConfigFlags.ToRawKubeConfigLoader().Namespace()
+		}
+
+		var listOptions = metav1.ListOptions{
+			LabelSelector: labels.Everything().String(),
+			FieldSelector: fields.Everything().String(),
+		}
+
+		nodes, err := client.CoreV1().Nodes().List(context.TODO(), listOptions)
+		if err != nil {
+			contextLogger.Fatalf("Error in listing nodes: %q", err)
+		}
+
+		restConfig, err := kubeRestConfig()
+		if err != nil {
+			contextLogger.Fatalf("Error while getting rest config: %s", err)
+		}
+
+		traceConfig := *restConfig
+		traceConfig.ContentConfig.GroupVersion = &gadgetv1alpha1.GroupVersion
+		traceConfig.APIPath = "/apis"
+		traceConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
+		traceConfig.UserAgent = restclient.DefaultKubernetesUserAgent()
+
+		traceRestClient, err := restclient.UnversionedRESTClientFor(&traceConfig)
+
+		for _, node := range nodes.Items {
+			if collectorParamNode != "" && node.Name != collectorParamNode {
+				continue
+			}
+
+			trace := &gadgetv1alpha1.Trace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: subCommand + "-",
+					Namespace:    "gadget",
+					Annotations: map[string]string{
+						GADGET_OPERATION: "start",
+					},
+					Labels: map[string]string{
+						"trace-template-hash": traceID,
+					},
+				},
+				Spec: gadgetv1alpha1.TraceSpec{
+					Node:   node.Name,
+					Gadget: subCommand,
+					Filter: &gadgetv1alpha1.ContainerFilter{
+						Namespace:     namespace,
+						Podname:       collectorParamPodname,
+						ContainerName: collectorParamContainername,
+						Labels:        labelsSelector,
+					},
+					RunMode:    "Manual",
+					OutputMode: "Status",
+				},
+			}
+			createdTrace := &gadgetv1alpha1.Trace{}
```

Comment:
What about using `.Error()` instead of `Into()` to avoid creating this unused variable?